buckets/buckets.go

```diff
@@ -171,18 +205,34 @@ func (b *Buckets) IsArchivingEnabled() bool {
 	return b.pgClient != nil
 }
 
-func createFFSInstance(ctx context.Context, col *collections.FFSInstances, c *powc.Client, bucketKey string) error {
+func (b *Buckets) createFFSInstance(ctx context.Context, bucketKey string) error {
 	// If the Powergate client isn't configured, don't do anything.
-	if c == nil {
+	if b.pgClient == nil {
 		return nil
 	}
-	_, token, err := c.FFS.Create(ctx)
+	_, token, err := b.pgClient.FFS.Create(ctx)
 	if err != nil {
 		return fmt.Errorf("creating FFS instance: %s", err)
 	}
-	if err := col.Create(ctx, bucketKey, token); err != nil {
+
+	ctxFFS := context.WithValue(ctx, powc.AuthKey, token)
+	i, err := b.pgClient.FFS.Info(ctxFFS)
+	if err != nil {
+		return fmt.Errorf("getting information about created ffs instance: %s", err)
+	}
+	waddr := i.Balances[0].Addr
+	if err := b.ffsCol.Create(ctx, bucketKey, token, waddr); err != nil {
 		return fmt.Errorf("saving FFS instances data: %s", err)
 	}
+	defaultBucketCidConfig := ffs.DefaultConfig{
+		Cold:       b.buckCidConfig.Cold,
+		Hot:        b.buckCidConfig.Hot,
+		Repairable: b.buckCidConfig.Repairable,
+	}
+	defaultBucketCidConfig.Cold.Filecoin.Addr = waddr
+	if err := b.pgClient.FFS.SetDefaultConfig(ctxFFS, defaultBucketCidConfig); err != nil {
+		return fmt.Errorf("setting default bucket FFS cidconfig: %s", err)
+	}
```

Comment:
Sounds good! I suppose the wallet address could be written to the actual bucket model, which would make the fact that a bucket is backed by FIL more transparent, but not sure if there are any security risks there.

Reply:
Well, a priori shouldn't be a big risk regarding if somebody know only your wallet address, can't do anything _per se_.
On some more paranoid mindset, maybe someone wouldn't like to expose that since having the wallet address means you can check how much balance is in there... so maybe that's preferred to be avoided?

We can think it a bit more, or see if people ask for this and add it easily.

Reply:
Cool, sounds good. They could always use the `--private` flag to hide the address if it was on the bucket itself.